homeassistant/helpers/script.py

```diff
@@ -896,6 +906,26 @@ async def _async_error_step(self):
         trace_set_result(error=error)
         raise _AbortScript(error)
 
+    @async_trace_path("parallel")
+    async def _async_parallel_step(self) -> None:
+        """Run a sequence in parallel."""
+        # pylint: disable=protected-access
+        scripts = await self._script._async_get_parallel_scripts(self._step)
+
+        async def async_run_with_trace(idx: int, script: Script) -> None:
+            """Run a script with a trace path."""
+            trace_path_stack_cv.set(copy(trace_path_stack_cv.get()))
```

Comment:
So I am not 100% sure about context vars, but do we need to set it to a copy? Doesn't `trace_path` already make a copy?

Reply:
It does, however, the path is set before the task... thus the second task will append on the stack. Since this is only an issue for the path stack, it's copied before doing that.